rickandmorty/src/characterdetails.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import PropTypes from "prop-types";
+import "./App.css";
+
+export default function CharacterDetails() {
+  const [character, characterDetail] = useState({});
+  const { id } = useParams();
+
+  useEffect(() => {
+    fetch(`https://rickandmortyapi.com/api/character/${id}`)
+      .then((res) => res.json())
+      .then((data) => characterDetail(data));
+  });
```

Comment:
Just for future reference, the example I showed in the week 5 code examples had a mistake. There should be an empty array as a second parameter for `useEffect`, otherwise it causes an infinite loop and keeps fetching the data unnecessariliy.

https://medium.com/@andrewmyint/infinite-loop-inside-useeffect-react-hooks-6748de62871